pkg/controllers/machinesetsync/machineset_sync_controller_test.go

```diff
@@ -156,56 +155,280 @@ var _ = Describe("With a running MachineSetSync controller", func() {
+				capiMachineSet = capiMachineSetBuilder.Build()
+
+				Expect(k8sClient.Create(ctx, capiMachineSet)).Should(Succeed())
+				Expect(k8sClient.Create(ctx, mapiMachineSet)).Should(Succeed())
+
+				By("Setting the AuthoritativeAPI to Migrating")
+				Eventually(k.UpdateStatus(mapiMachineSet, func() {
+					mapiMachineSet.Status.AuthoritativeAPI = machinev1beta1.MachineAuthorityMigrating
+				})).Should(Succeed())
+			})
+
+			It("should not make any changes", func() {
+				resourceVersion := mapiMachineSet.GetResourceVersion()
+				Consistently(k.Object(mapiMachineSet), timeout).Should(
+					HaveField("ResourceVersion", Equal(resourceVersion)),
+				)
+			})
+		})
+
+		Context("When the MAPI MachineAuthority is not set", func() {
+			BeforeEach(func() {
+				By("Creating the CAPI and MAPI MachineSets")
+				mapiMachineSet = mapiMachineSetBuilder.Build()
+				capiMachineSet = capiMachineSetBuilder.Build()
+
+				Expect(k8sClient.Create(ctx, capiMachineSet)).Should(Succeed())
+				Expect(k8sClient.Create(ctx, mapiMachineSet)).Should(Succeed())
+
+				By("Setting the AuthoritativeAPI to Migrating")
+				Eventually(k.UpdateStatus(mapiMachineSet, func() {
+					mapiMachineSet.Status.AuthoritativeAPI = ""
+				})).Should(Succeed())
+			})
+
+			// I think there may be another case here, where we have had it previously set - synchronized and then it gets unset.
+			// We may want to ensure that we don't change resourceVersion, and that the condition does not change?
+			// Changes to false with a failure reason?
+			//
+			// I'm not sure if we have an API validation that stops this.
+			// TODO: clarify / work out
+
+			It("should not make any changes", func() {
+				resourceVersion := mapiMachineSet.GetResourceVersion()
+				Consistently(k.Object(mapiMachineSet), timeout).Should(
+					HaveField("ResourceVersion", Equal(resourceVersion)),
+				)
+			})
+		})
+
+		Context("when the MAPI MachineSet does not exist and the CAPI MachineSet does", func() {
+			BeforeEach(func() {
+				By("Creating the CAPI MachineSet")
+				capiMachineSet = capiMachineSetBuilder.Build()
+
+				Expect(k8sClient.Create(ctx, capiMachineSet)).Should(Succeed())
+			})
+
+			It("should not make any changes to the CAPI MachineSet", func() {
+				resourceVersion := capiMachineSet.GetResourceVersion()
+				Consistently(k.Object(capiMachineSet), timeout).Should(
+					HaveField("ResourceVersion", Equal(resourceVersion)),
+				)
+			})
+
+			// Do we have another case here where we consistently want to not see a MAPI MachineSet created?
+			// Not sure if this is overkill
+
+		})
+
 	})
+
+	Context("when the infrastructure resources don't exist", func() {
+		Context("when the MAPI machine set has MachineAuthority set to Cluster API", func() {
+			BeforeEach(func() {
+				By("Creating the CAPI and MAPI MachineSets")
+				mapiMachineSet = mapiMachineSetBuilder.Build()
+				capiMachineSet = capiMachineSetBuilder.WithReplicas(int32(4)).Build()
```

Comment:
Why are we not checking that the replicas count is copied over? I agree not to do in-depth testing of the conversion since that's handled elsewhere, but, this seems like an easy smoke test to show that it is actually employing conversion

Reply:
I think we do further up - I just forgot to change when copypasta: 

```go
				It("should update the replica count", func() {
					Eventually(k.Object(mapiMachineSet), timeout).Should(
						HaveField("Spec.Replicas", Equal(ptr.To(int32(4)))),
					)
				})
```